app/controllers/declaration/RoutingCountriesController.scala

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.declaration
+
+import controllers.actions.{AuthAction, JourneyAction}
+import controllers.navigation.Navigator
+import forms.declaration.RoutingQuestionYesNo
+import forms.declaration.RoutingQuestionYesNo._
+import forms.declaration.destinationCountries.DestinationCountries
+import forms.declaration.destinationCountries.DestinationCountries.{FirstRoutingCountryPage, NextRoutingCountryPage}
+import javax.inject.Inject
+import models.{ExportsDeclaration, Mode}
+import models.requests.JourneyRequest
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import services.Countries.retrieveCountryNameFromCode
+import services.cache.ExportsCacheService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import views.html.declaration.destinationCountries.{country_of_routing, routing_country_question}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class RoutingCountriesController @Inject()(
+  authenticate: AuthAction,
+  journeyType: JourneyAction,
+  override val exportsCacheService: ExportsCacheService,
+  navigator: Navigator,
+  mcc: MessagesControllerComponents,
+  routingQuestionPage: routing_country_question,
+  countryOfRoutingPage: country_of_routing
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport with ModelCacheable {
+
+  def displayRoutingQuestion(mode: Mode, fastForward: Boolean): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
```

Comment:
fast forward?

Reply:
Nope, the same implementation is on different page. Different behaviour if user use back button. Idea taken from @SteveSugden

Reply:
Just so I understand correctly, the reason we need this is because the back button needs to know where to go to?

Cant we just work that out based on the user answers?

Reply:
@ed0906 When we have single page that could be reached from two sources where reach second depends on response it is easier to redirect with earlier source and left decision there.